primitives/messages/src/source_chain.rs

```diff
@@ -125,10 +125,22 @@ pub trait MessagesBridge<Payload> {
 	/// Error type.
 	type Error: Debug;
 
+	/// Intermediary structure returned by `validate_message()`.
+	///
+	/// It can than be passed to `send_message()` in order to actually send the message
+	/// on the bridge.
+	type SendMessageArgs;
```

Comment:
Let's probably remove this associated type and use a fixed `struct ValidatedMessage` (or something like that?). There's a lot of work done in the `master` branch to remove extra generic args/associated types in favor of fixed types. Or is there really a need for that?

Reply:
I added this in order to be able to define the actual `SendMessageArgs` struct inside the messages module. This way it doesn't have to expose a constructor. It can only be constructed there. So that it can't be created manually without going through the `validate_message()` function. WDYT ?

Reply:
@svyatonik sorry for the ping. WDYT ? Would you have any concerns about keeping this associated type ?

Reply:
Oh, sorry - missed that comment. Yeah - that makes sense. I think both options are fine in that case - leave as is, or change